### PR TITLE
Class review

### PR DESCRIPTION
**Pull request by torvista**
No changes to functionality. All based on IDE code inspections/complaints (based on php 7.1), intended to make real code changes later easier.
Code is in production on php8 with no issues with strict checks.

I intend to do the other files (that require far less work) prior to cleaning up the use of the width/heights integers/floats etc. that are scattered about liberally but need more examination.

### CODE REVIEW

**Comment by lat9** [comment]
Phew!  That's a slew of changes; I'll need some time to review.  Given that a lot of IH users are still (I know) on PHP 5.6, let's not go down the type-hinting path just yet to enable a wider audiences' use.

**Comment by torvista** [comment]
> PHP 5.6
Good grief, and I thought I was being conservative with 7.1!

I find adding all these restrictions usually chases out something from under a stone, so they are worth it for development.

I'll reduce it to 5.6 level.

**Review by lat9 — CHANGES_REQUESTED** [review]
Just like my other comment, I think we need to explore **why** those values aren't integer values coming in and correct any issue at the source.

**Review by lat9 — CHANGES_REQUESTED** [review]
Same comment, find the source of the issue instead of 'correcting' at the usage.

**Review by lat9 — CHANGES_REQUESTED** [review]
Again, need to find the source of the !integer values, but I'll suggest removing the `@` since it's going to be useless with later versions of PHP anyhoo.

File: 1_Installation_Files (v1.5.5)/includes/classes/bmz_image_handler.class.php [hunk 1]
```diff
@@ -864,6 +864,8 @@ protected function load_imageGD($src_name)
                 }
                 $image = @imagecreatefromjpeg($src_name);
                 break;
+            default:
+                $image = false;
```
Comment (lat9) [inline 1_Installation_Files (v1.5.5)/includes/classes/bmz_image_handler.class.php:957]: Please add a `break;` at the end of the default clause.

**Comment by lat9** [comment]
Let me know when you've finished; we'll get any other changes after retesting after the merge.

**Comment by torvista** [comment]
Done.

**Pull request merged**